biz.aQute.bndlib/src/aQute/bnd/osgi/Contracts.java

```diff
@@ -166,7 +179,21 @@ void addToRequirements(Parameters requirements) {
 
 			Formatter f = new Formatter();
 			try {
-				f.format("(&(%s=%s)(version=%s))", ContractNamespace.CONTRACT_NAMESPACE, c.name, c.version);
+				String versionString = "(version=%s)";
+				if (c.versions.size() == 1) {
+					versionString = String.format(versionString, c.versions.get(0));
+				} else {
+					StringBuilder sb = new StringBuilder("(|");
+					for (Version version : c.versions) {
+						sb.append("(version=");
+						sb.append(version);
+						sb.append(")");
+					}
+					sb.append(")");
+					versionString = sb.toString();
+				}
+
+				f.format("(&(%s=%s)%s)", ContractNamespace.CONTRACT_NAMESPACE, c.name, versionString);
```

Comment:
right, I wasn't quite sure about it that's why I went this approach. So will the logic work if we simply make N clones of the Contract one per version?